rpm/ndb/package.go

```diff
@@ -42,7 +42,7 @@ func (db *PackageDB) Parse(r io.ReaderAt) error {
 	db.slot = make([]pkgSlot, 0, lastIndex)
 	b = b[:slotSize]
 	// Read every populated slot (these should be contiguous) and populate the lookup table.
-	for off := int64(slotStart * slotSize); ; off += slotSize {
+	for off := int64(slotStart * slotSize); off < int64(db.NPages*pageSz); off += slotSize {
```

Comment:
So this works by virtue of reading the whole page and hitting the end. I suppose the extra time to read a few K isn't the end of the world.